src/test/java/system_tests/TrialSeleniumTest.java

```diff
@@ -1,73 +1,208 @@
 
+    // Check for decoder
+    WebElement decodersTable = driver.findElement(By.tagName("table")); // find encoders table
+    List<WebElement> devicesRows =
+        decodersTable.findElements(By.tagName("tr")); // find all tr elements inside found table
     boolean assertValue =
-        devicesRows.stream().anyMatch(row -> row.getText().contains(deviceEntity.getDisplayName()));
+        devicesRows.stream()
+            .anyMatch(
+                row ->
+                    row.getText()
+                        .contains(DecoderFixture.getDecoderEntity2().getDevice().getDisplayName()));
 
     assertTrue(assertValue);
   }
 
-  @AfterEach
-  void tearDown() {
-    driver.quit();
+  @Test
+  @Order(3)
+  void testCreateStream() {
+    driver.get("http://localhost:3000/Streaming");
+    {
+      WebDriverWait wait = new WebDriverWait(driver, 100);
+      wait.until(
+          ExpectedConditions.elementToBeClickable(
+              By.cssSelector("#SenderTable .MuiList-root .MuiSvgIcon-root")));
+    }
+    driver.findElement(By.cssSelector("#SenderTable .MuiList-root .MuiSvgIcon-root")).click();
+    {
+      WebElement element;
+      WebDriverWait wait = new WebDriverWait(driver, 100);
+      element =
+          wait.until(ExpectedConditions.elementToBeClickable(By.cssSelector(".MuiInput-input")));
+      Actions builder = new Actions(driver);
+      builder.moveToElement(element).clickAndHold().perform();
+    }
+    {
+      WebElement element;
+      WebDriverWait wait = new WebDriverWait(driver, 100);
+      element =
+          wait.until(ExpectedConditions.elementToBeClickable(By.cssSelector(".MuiMenuItem-root")));
+      Actions builder = new Actions(driver);
+      builder.moveToElement(element).release().perform();
+    }
+    driver.findElement(By.cssSelector("body")).click();
+    driver.findElement(By.cssSelector(".MuiMenuItem-root")).click();
+    driver.findElement(By.cssSelector("#ReceiverTable .MuiList-root .MuiSvgIcon-root")).click();
+    {
+      WebElement element;
+      WebDriverWait wait = new WebDriverWait(driver, 100);
+      element =
+          wait.until(
+              ExpectedConditions.elementToBeClickable(
+                  By.xpath(
+                      "//div[@id='ReceiverTable']/div[3]/ul/div[2]/div/div/div/li/div[2]/div")));
+      Actions builder = new Actions(driver);
+      builder.moveToElement(element).clickAndHold().perform();
+    }
+    {
+      WebElement element;
+      WebDriverWait wait = new WebDriverWait(driver, 100);
+      element =
+          wait.until(ExpectedConditions.elementToBeClickable(By.cssSelector(".MuiMenuItem-root")));
+      Actions builder = new Actions(driver);
+      builder.moveToElement(element).release().perform();
+    }
+    driver.findElement(By.cssSelector("body")).click();
+    driver.findElement(By.cssSelector(".MuiMenuItem-root")).click();
+    driver.findElement(By.cssSelector(".buttonText")).click();
```

Comment:
I see that this test doesn't have any asserts, could we call the `/stream` endpoint and check that it doesn't return an empty list maybe?

Reply:
Yeah, it’s because the create stream and view stream are two separate tests, but I think they might as well be combined into one. The view test is pretty much the assert of the create test.

Reply:
I wouldn't mind combining them into one test